kinode/packages/app_store/app_store/src/lib.rs

```diff
@@ -329,577 +129,334 @@ fn handle_message(
-        };
-        let Some(blob) = get_blob() else {
-            return (LocalResponse::GetApiResponse(GetApiResponse::Failure), None);
-        };
-        let blob = LazyLoadBlob {
+pub fn get_api(state: &mut State, package_id: &PackageId) -> (LocalResponse, Option<LazyLoadBlob>) {
+    if !state.downloaded_apis.contains(package_id) {
+        return (LocalResponse::GetApiResponse(GetApiResponse::Failure), None);
+    }
+    let Ok(Ok(_)) = Request::new()
+        .target(("our", "vfs", "distro", "sys"))
+        .body(
+            serde_json::to_vec(&vfs::VfsRequest {
+                path: format!("/{package_id}/pkg/api.zip"),
+                action: vfs::VfsAction::Read,
+            })
+            .unwrap(),
+        )
+        .send_and_await_response(VFS_TIMEOUT)
+    else {
+        return (LocalResponse::GetApiResponse(GetApiResponse::Failure), None);
+    };
+    let Some(blob) = get_blob() else {
+        return (LocalResponse::GetApiResponse(GetApiResponse::Failure), None);
+    };
+    (
+        LocalResponse::GetApiResponse(GetApiResponse::Success),
+        Some(LazyLoadBlob {
             mime: Some("application/json".to_string()),
             bytes: blob.bytes,
-        };
-        (GetApiResponse::Success, Some(blob))
-    };
-    (LocalResponse::GetApiResponse(response), blob)
+        }),
+    )
 }
 
 pub fn list_apis(state: &mut State) -> LocalResponse {
-    LocalResponse::ListApisResponse {
-        apis: state.downloaded_apis.iter().cloned().collect(),
-    }
+    LocalResponse::ApisResponse(ApisResponse {
+        apis: state
+            .downloaded_apis
+            .clone()
+            .into_iter()
+            .map(|id| crate::kinode::process::main::PackageId::from_process_lib(id))
+            .collect(),
+    })
 }
 
-pub fn rebuild_index(
-    our: &Address,
-    state: &mut State,
-    eth_provider: &eth::Provider,
-    requested_apis: &mut HashMap<PackageId, RequestedPackage>,
-) -> LocalResponse {
-    *state = State::new(CONTRACT_ADDRESS.to_string()).unwrap();
+pub fn rebuild_index(state: &mut State) -> LocalResponse {
     // kill our old subscription and build a new one.
-    let _ = eth_provider.unsubscribe(1);
-
-    let filter = eth::Filter::new()
-        .address(eth::Address::from_str(&state.contract_address).unwrap())
-        .from_block(state.last_saved_block - 1)
-        .events(EVENTS);
+    let _ = state.provider.unsubscribe(1);
 
-    subscribe_to_logs(&eth_provider, filter.clone());
-
-    for log in fetch_logs(&eth_provider, &filter) {
-        if let Err(e) = state.ingest_listings_contract_event(our, log, requested_apis) {
-            println!("error ingesting log: {e:?}");
-        };
-    }
+    let eth_provider = eth::Provider::new(CHAIN_ID, CHAIN_TIMEOUT);
+    *state = State::new(
+        state.our.clone(),
+        eth_provider,
+        state.contract_address.clone(),
+    )
+    .expect("state creation failed");
```

Comment:
Is there a good reason this isn't `?`d? I see the callers of it would need to be modified to return `Result`s, but its not obvious to me there is a good reason to not do that. If we don't have a good reason to not `?` then we should `?`.

Reply:
my reasoning is that the app store will fail in complex and hard to reason about ways if the `populate_packages_from_filesystem` call within `State::new()` fails for any reason

Reply:
Seems reasonable